lib/wallet/src/Cardano/Wallet/Delegation/Model.hs

```diff
@@ -58,10 +75,56 @@ instance (Ord slot, Eq pool) => Delta (Operation slot pool) where
         miss = status slot hist'
         wanted = transition r $ status slot hist
 
+--                                              DelegateAndVote
+--                                                     |
+--                                           Delegate  |   Vote
+--                                               |     |    | -------
+--                                               v     v    v |     |
+--                     ------------  Register   ---------------     | Register
+--                     | Inactive |------------>|  Registered |------
+--                     ------------             ---------------
+--                                 ----------    |     |    |    ---------
+--                                 |        v    v     |    v    |       |
+--                                 |       ----------  |  ---------      | Vote
+--                        Delegate --------| Active |  |  | Voted |<------
+--                                         ----------  |  ---------
+--                                           Vote |    |      | Delegate
+--                                DelegateAndVote v    v      v DelegateAndVote
+--                                               ------------------
+--                                               | ActiveAndVoted |<---
+--                                               ------------------   | Delegate
+--                                                         |          | DelegateAndVote
+--                                                         ------------ Vote
+--
+--
+--                     ------------ Deregister  ---------------
+--                     | Inactive |<----------- |  Registered |
+--                     ------------             ---------------
+--                    /\ /\ /\ /\
+--                     |  |  |  | Deregister
+--                     |  |  |  |       ----------     ---------
+--                     |  |  |  --------| Active |     | Voted |
+--                     |  |             ----------     ---------
+--                     |  |        Deregister             |
+--                     |  ---------------------------------
+--                     |                         ------------------
+--                     --------------------------| ActiveAndVoted |
+--                             Deregister        ------------------
+--
```

Comment:
Please remove the diagrams from the source code and add them to the specification file https://github.com/cardano-foundation/cardano-wallet/blob/master/specifications/Cardano/Wallet/delegation.md instead.